include/zombie/kinetic.hpp

```diff
@@ -316,27 +324,35 @@ struct MinHanger : MinHeapCRTP<T, MinHanger<T, Compare, NHIC, NHER>> {
     return size() == 0;
   }
 
-  void hang(T& t, const size_t& idx) {
+  void hang(T&& t, const size_t& idx) {
     if (!(idx < arr.size())) {
       arr.resize(idx + 1);
     }
     if (arr[idx].has_value()) {
       // TODO: huh, is 'prioritizing an empty child' a worthwhile optimization?
       size_t child_idx = heap_left_child(idx) + (coin() ? 0 : 1);
       if (cmp(t, arr[idx].value())) {
-        std::swap(t, arr[idx].value());
+        T t0 = std::forward<T>(arr[idx].value());
```

Comment:
you should be moving. forwarding is for passing a function's template argument to another function, as it handle both reference and rreference. essentially speaking I dont think any code here should be forwarding.